util/topgen/validate.py

```diff
@@ -259,6 +298,99 @@
 reset_connection_optional = {}
 reset_connection_added = {}
 
+reset_requests_required = {}
+reset_requests_optional = {
+    'int': ['s', 'internal request list'],
+    'debug': ['s', 'debug request list'],
+    'peripheral': ['s', 'peripheral request list'],
+}
+reset_requests_added = {}
+
+reset_request_required = {
+    'name': ['s', 'the reset request name'],
+    'desc': ['s', 'the reset request descrption'],
+    'module': ['s', 'the reset request source'],
+}
+reset_request_optional = {
+    'width': ['d', 'TODO'],
+}
+reset_request_added = {}
+
+wakeup_required = {
+    'name': ['s', 'the wakeup name'],
+    'width': ['d', 'the width of the signal'],
+    'module': ['s', 'the module sourcing the wakeup'],
+}
+wakeup_optional = {}
+wakeup_added = {}
+
+alert_required = {
+    'name': ['s', 'name of the alert signal'],
+    'width': ['d', 'the number of alerts in this signal, typically 1'],
+    'async': ['s', 'string interpreted as boolean'],
+    'module_name': ['s', 'The module name of the source'],
+}
+alert_optional = {
+    'desc': ['s', 'the description of the alert'],
+    'lpg_name': ['s', 'the low power group of the alert'],
+    'lpg_idx': ['d', 'the index in the lpg group'],
+    'type': ['s', 'should contain "alert"'],
+}
+alert_added = {}
+
+interrupt_required = {
+    'name': ['s', 'the name of the interrupt'],
+    'width': ['d', 'the number of interrupts in this signal, typically 1'],
+    'module_name': ['s', 'The module name of the source'],
+    'intr_type': ['s', 'The IntrType, either Event or Status'],
+    'default_val': ['s', 'a string interpreted as boolean'],
+    'incoming': ['s', 'a string interpreted as boolean'],
+}
+interrupt_optional = {
+    'desc': ['s', 'the description of the interrupt'],
+    'type': ['s', 'should contain "interrupt"'],
+}
+interrupt_added = {}
+
+param_required = {
+    'name': ['s', 'the parameter name'],
+    'desc': ['s', 'the parameter description'],
+    'type': ['s', 'the data type of the parameter'],
+    'default': ['s', 'the default value of the parameter'],
+}
+param_optional = {
+    'expose': ['s', 'seems redundant TODO'],
+    'local': ['s', 'whether it is a localparam, interpreted as boolean'],
+    'name_top': ['s', 'the name in the top-level'],
+    'randcount': ['d', 'TODO'],
```

Comment:
This is apparently the number of bits requested for a random number inserted into toplevel_rnd_cnst_pkg.sv. If I'm not misreading the code, it looks like `randwidth` is added ...and is just equal to `randcount`.

Reply:
I think they are not always the same, which is what triggered the TODO. I think we need a separate PR which can generate documentation about the format of the hjson files partly based on the key's documentation.